### ISSUE TITLE
`auto a=getA();` works; `A a=getA();` fails to compile (with A getA())

### ISSUE THREAD

**Issue by timotheecour** [Syniurge/Calypso#72]
BUG:D20180112T164945
```
C++:
A getA();

D:
auto a=getA();//works
static assert(is(typeof(a)==A)); //works

A a=getA();//fails:
// Error: none of the overloads of '__ctor' are callable using argument types (), 
```

Likewise:
```
auto a=A.init; // does not call any C++ functions
A a=A.init; // calls A::A, A::A(A&&), A::~A
```

this seems wrong; better semantics should be:

* `auto a=` is equivalent to `A a` whenever `typeof(a)==A`
* `A a=A.init;` and `A a` shouldn't call the c++ constructor `A::A()`, instead just initialize a to compile-time known A.init by memcpy
* `A a=A();` (or `auto a=A();`) should call C++ `A::A()`

**Comment by Syniurge** [comment]
Hmm yes, that's a bug. The copy constructor support has been messy, initially when I started working on Calypso D still had copy ctors so this was plug and play. But then they got replaced by postblits and the copy ctor code gradually disappeared from DMD/LDC so I had to gradually replace it by my own. This needs to get fixed asap.

**Comment by Syniurge** [comment]
Should be fixed by 59c119649cd525949d0a2842a8a52972b1e15369 and d5bcdf99da90bc126896aa54f7d61e20d9d969d8.
